client/src/persistent_receiver.rs

```diff
@@ -0,0 +1,76 @@
+use tokio::sync::{broadcast, mpsc};
+
+pub struct PersistentReceiver<T>(mpsc::UnboundedReceiver<T>);
+
+impl<T: Clone + Send + 'static> PersistentReceiver<T> {
+    pub fn new<F>(mut r: broadcast::Receiver<T>, filter: Option<F>) -> Self
+    where
+        F: Fn(&T) -> Option<T> + Send + Sync + 'static,
+    {
```

Comment:
Forslag: brug `where` til `T` også

```suggestion
impl<T> PersistentReceiver<T> {
    pub fn new<F>(mut r: broadcast::Receiver<T>, filter: Option<F>) -> Self
    where
        T: Clone + Send + 'static,,
        F: Fn(&T) -> Option<T> + Send + Sync + 'static,
    {
```